cmd/prometheus/main.go

```diff
@@ -713,6 +714,36 @@ func main() {
 			},
 		)
 	}
+
+	// Watch the config file for changes
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		level.Error(logger).Log("failed to create watcher", "err", err)
+		os.Exit(1)
+	}
+
+	go func() {
+		for {
+			select {
+			case <-watcher.Events: // reload on all events
+				if err := reloadConfig(cfg.configFile, logger, reloaders...); err != nil {
+					level.Error(logger).Log("failed to reload config", "err", err)
+					os.Exit(1)
```

Comment:
I don't think you want to exit 1 here. If reload fails just use the old config. [reference](https://github.com/digitalocean/prometheus/blob/420246938db44a38d2da85020b59dc514016811e/cmd/prometheus/main.go#L566)